assets/src/js/Components/ArchivePicker.js

```diff
@@ -0,0 +1,151 @@
+    };
+    this.loadNextPage = this.loadNextPage.bind( this );
+    this.updateFromUploadedResponse = this.updateFromUploadedResponse.bind( this );
+    this.search = this.search.bind( this );
+    this.includeInWp = this.includeInWp.bind( this );
+  }
+
+  async componentDidMount() {
+    await this.loadNextPage();
+  }
+
+  async fetchImages( args ) {
+    return apiFetch( {
+      path: addQueryArgs( '/planet4/v1/image-archive/fetch', args ),
+    } );
+  }
+
+  async loadNextPage( newSearch = false ) {
+    this.setState( { loading: true } );
+
+    const pageIndex = newSearch ? 1 : this.state.currentPage + 1;
+    const searchedText = newSearch ? this.state.enteredSearch : this.state.searchText;
+
+    try {
+      const nextImages = await this.fetchImages( {
+        page: pageIndex,
+        search_text: searchedText,
+      } );
+      const withPageLabel = nextImages.map( image => ( {
+        ...image,
+        pagedTitle: `${ pageIndex } -- ${ image.title }`
+      } ) );
+      this.setState( {
+        currentPage: pageIndex,
+        images: [
+          ...( newSearch ? [] : this.state.images ),
+          ...withPageLabel
+        ],
+        searchText: searchedText,
+      } );
+    } catch ( error ) {
+      this.setState( { error } );
+    } finally {
+      this.setState( { loading: false } );
+    }
+  }
+
+  async includeInWp( ids ) {
+    try {
+      this.setState( { processingImages: true } );
+      const updatedImages = await apiFetch( {
+        method: 'POST',
+        path: '/planet4/v1/image-archive/transfer',
+        data: {
+          ids: ids,
+          use_original_language: false,
+        }
+      } );
+      this.updateFromUploadedResponse( updatedImages );
+    } catch ( e ) {
+      console.log( e );
+      this.setState( { processingError: e } );
+    } finally {
+      this.setState( { processingImages: false } );
+    }
+  }
+
+  updateFromUploadedResponse( updatedImages ) {
+    const newImages = this.state.images.map( stateImage => {
+      const updated = updatedImages.find( updatedImage => updatedImage.id === stateImage.id );
+      if ( updated ) {
+        return updated;
+      }
+      return stateImage;
+    } );
+    this.setState( {
+      images: newImages,
+    } );
+  }
+
+  async search() {
+    await this.loadNextPage( true );
+  }
+
+  render() {
+    const {
+      loading,
+      error,
+      images,
+    } = this.state;
+
+    return <Fragment>
+      <form
+        className={ 'archive-picker-search' }
+        onSubmit={ event => {
+          event.preventDefault();
+          this.search();
+        } }
+        onChange={ event => this.setState( { enteredSearch: event.target.value } ) }
+      >
+        <input
+          type={ 'text' }
+          disabled={ this.state.loading }
+        />
+        <input type={ 'submit' } value={ __( 'Search' ) }/>
+      </form>
+      { loading && (
+        <div className={ 'archive-picker-loading' }> loading...</div>
```

Comment:
need string translation

Reply:
Maybe for now we should wait with the string translation until we have the final text? Since the feature is not intended to be used live yet, we might make changes to these texts before they are ever used. So seems to make more sense to only add strings to the translations when we know they will be used, to avoid unnecessary translation work.